### PR TITLE
upgrade instruction set

### PR DESCRIPTION
**Pull request by thempp66**

### CODE REVIEW

File: src/processor.cpp [hunk 1]
```diff
@@ -602,19 +622,29 @@ class _Processor
             case Opcode::TIMESTAMP:
                 timestamp();
                 break;
-            case Opcode::DIFFICULTY:
-                difficulty();
+            case Opcode::PREVRANDAO:
+                prevrandao();
```
Comment (yingjun-wu) [inline src/processor.cpp:626]:
```suggestion
                prevrandao();
```
is difficult?

Comment (thempp66) [reply src/processor.cpp:626]: Renaming: The DIFFICULTY (0x44) opcode SHOULD further be renamed to PREVRANDAO (0x44).

File: src/processor.cpp [hunk 2]
```diff
@@ -1162,21 +1251,39 @@ class _Processor
...
-        const auto newAddress = generate_address(ctxt->acc.get_address(), ctxt->acc.get_nonce());
+        Address newAddress;
+        if (op == Opcode::CREATE)
+        {        
+            newAddress = generate_address(ctxt->acc.get_address(), ctxt->acc.get_nonce());
+        } 
+        else if (op == Opcode::CREATE2)
+        {
+            const auto salt = ctxt->s.pop();
+            newAddress = generate_address_for_create2(ctxt->acc.get_address(), salt, contractValue);
```
Comment (yingjun-wu) [inline src/processor.cpp:1285]: in create2, contractValue should be initCode

File: src/util.cpp [hunk 3]
```diff
@@ -50,6 +50,31 @@ namespace eevm
         return from_big_endian(Keccak256(rlp_encoding).data() + 12u, 20u);
     }
 
+    Address generate_address_for_create2(const Address& sender, uint256_t salt, uint256_t init_code)
```
Comment (yingjun-wu) [inline src/util.cpp:53]: init_code variable should be std::vector<uint8_t>, and then keccak256 after use it

File: src/processor.cpp (hunk 2 above)
Comment (thempp66) [reply src/processor.cpp]: fixed

File: src/util.cpp (hunk 3 above)
Comment (thempp66) [reply src/util.cpp]: fixed

File: src/processor.cpp (hunk 1 above)
Comment (yingjun-wu) [reply src/processor.cpp:626]: Since the contract execute on off-chain, it is not necessary for some opcode, such as PREVERANDAO, BASEFEE, GASLIMIT, etc.

Comment (thempp66) [reply src/processor.cpp:627]: Done

**Review by yingjun-wu — APPROVED** [review]

**Pull request merged**